File: src/main/java/game/Peca.java
package game;

public abstract class Peca {
    public String nomePeca;

    public Peca(String cor) {
        this.cor = cor;
    }

    protected String cor;

    abstract public boolean isMovimentoValido(Tabuleiro tabuleiro, int linhaInicial, int colunaInicial, int linhaFinal, int colunaFinal);

    public String getNomePeca() {
        return nomePeca;
    }

    public String getCor() {
        return cor;
    }


}

File: Projeto JS/src/game/Peca.js
export class Peca {
    constructor(cor) {
        this.cor = cor;
        this.nomePeca = "";
    }

    getCor() {
        return this.cor;
    }

    getNomePeca() {
        return this.nomePeca;
    }

    isMovimentoValido(tabuleiro, linhaInicial, colunaInicial, linhaFinal, colunaFinal) {
        throw new Error("O método 'isMovimentoValido' precisa ser implementado pela subclasse.");
    }
}